app/api_1_0/url.py

```diff
@@ -0,0 +1,399 @@
+    """
+    Return a shortened URL details after saving it along its long URL.
+
+    Keyword arguments:
+    new_short_url -- the newly generated shortened URL
+    long_url -- the new long URL to be linked and saved with the shortened URL
+    """
+    new_short_url.save()
+    new_long_url = LongUrl(long_url=long_url)
+    new_long_url.short_urls.append(new_short_url)
+    g.current_user.short_urls.append(new_short_url)
+    new_long_url.save()
+    return jsonify({'success': True,
+                    'url': new_short_url.get_details()
+                    })
+
+
+def generate_short_url(len_range=5):
+    """
+    This is a recursive function that generates a random string of length
+    len_range.
+
+    Keyword arguments:
+    len_range -- the integer value used to determine the length of the string
+                 it is given a default value of 5
+    """
+    url_prefix = dotenv.get('URL_PREFIX')
+    new_short_url = url_prefix + ''.join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase
+                      + string.digits) for _ in range(len_range))
+    if ShortUrl.query.filter_by(short_url=new_short_url).first():
+        generate_short_url(len_range + 1)
+    return ShortUrl(short_url=new_short_url)
+
+
+def get_short_url():
+    """
+    This function returns a short_url object according to the user and/or the
+    the inputs supplied through the request.
+
+    Keyword arguments:
+    vanity -- a string value that is sent through the request body
+    """
+    new_short_url = None
+    if request.json.get('vanity') and not g.current_user.is_anonymous:
+        new_short_url = get_vanity_url(request.json.get('vanity'))
+        return new_short_url
+    if request.json.get('vanity') and g.current_user.is_anonymous:
+        abort(unauthorized('Invalid credentials'))
+    new_short_url = generate_short_url()
+    return new_short_url
+
+
+def get_vanity_url(vanity):
+    """
+    This function returns a shortened URL generated using vanity string
+
+    Keyword arguments:
+    vanity -- a string value that is sent through the request body
+    """
+    url_prefix = dotenv.get('URL_PREFIX')
+    full_vanity_url = url_prefix + vanity
+    if ShortUrl.query.filter_by(
+            short_url=full_vanity_url).filter_by(deleted=0).first():
+        vanity_error = "Vanity string '{}' has been taken".format(vanity)
+        abort(forbidden(vanity_error))
+    return ShortUrl(short_url=full_vanity_url)
+
+
+@api.route('/shorturl/recent', methods=['GET'])
+def most_recent():
+    """
+    This function returns the list of most recently added shortened URLs.
+    """
+    recent_urls = ShortUrl.query.order_by(
+        desc(ShortUrl.date_time)).filter_by(deleted=0).limit(10).all()
+    if recent_urls:
+        urls = []
+        for i in range(len(recent_urls)):
+            urls.append(recent_urls[i].get_details())
+        return jsonify({'success': True, 'recents': urls})
+    return not_found('No url found')
```

Comment:
which `url` was not found here? the url i.e. the endpoint or the url we were looking for from the endpoint? More descriptive errors please..